kratos/utilities/auxiliar_model_part_utilities.cpp

```diff
@@ -83,8 +84,8 @@ void AuxiliarModelPartUtilities::EnsureModelPartOwnsProperties(const bool Remove
         }
 
         // Combine buffers together
-        #pragma omp critical
         {
+            ScopeLock lock;
```

Comment:
i think that the LockObject does initialization at creation time, so it cannot be uninitialized

Reply:
The order is the following:
ScopeLock:
- CTor: Set Lock (done via its `static` `LockObject` with `SetLock`)
- DTor: Free Lock (done via its `static` `LockObject` with `UnSetLock`)

The internal static `LockObject` is initialized once (where the actual (e.g. omp) lock is created)